source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -15,25 +15,93 @@
 #include <maya/MFnTypedAttribute.h>
 #include <maya/MGlobal.h>
 
+#include <maya/MViewport2Renderer.h>
+#include <maya/MFragmentManager.h>
+
 #include <algorithm>
 #include <sstream>
 
+namespace mx = MaterialX;
+
 namespace
 {
-    const char* const kDocumentFlag     = "d";
-    const char* const kDocumentFlagLong = "document";
+const char* const kDocumentFlag     = "d";
+const char* const kDocumentFlagLong = "document";
+
+const char* const kElementFlag      = "e";
+const char* const kElementFlagLong  = "element";
+
+const char* const kTextureFlag      = "t";
+const char* const kTextureFlagLong  = "texture";
+
+const char* const kOgsXmlFlag       = "x";
+const char* const kOgsXmlFlagLong   = "ogsxml";
+
+void registerFragment(const MaterialXData& materialData, const std::string& ogsXmlPath)
+{
+    MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
+    MHWRender::MFragmentManager* fragmentManager = theRenderer ? theRenderer->getFragmentManager() : nullptr;
+    if (!fragmentManager)
+    {
+        return;
+    }
+
+    MString previousOutputDirectory(fragmentManager->getEffectOutputDirectory());
+    MString previousIntermdiateDirectory(fragmentManager->getIntermediateGraphOutputDirectory());
+    mx::FilePath dumpPath(Plugin::instance().getShaderDebugPath());
+    bool setDumpPath = !dumpPath.isEmpty();
+    if (setDumpPath)
+    {
+        fragmentManager->setEffectOutputDirectory(dumpPath.asString().c_str());
+        fragmentManager->setIntermediateGraphOutputDirectory(dumpPath.asString().c_str());
+    }
+
+    // Register fragments with the manager if needed
+    const std::string& fragmentString = materialData.getFragmentWrapper();
+    const std::string& fragmentName = materialData.getFragmentName();
+    if (fragmentName.empty() || fragmentString.empty())
+    {
+        return;
```

Comment:
Should this throw an exception instead?

Reply:
I'll rework this a bit. I want to work on a catch-all for a "fallback" shader so this should not even return here.